### PR TITLE
System Info: Fix - "recalculate" button throw a system error.

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: modules/system-info/assets/js/admin/module.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		super.onInit();
+
+		// Hack to remove the query params after recalculate the elementor usage.
+		this.removeRecalcUsageQueryParam();
+	}
+
+	/**
+	 * Removes recalculate params from the url.
+	 */
+	removeRecalcUsageQueryParam() {
```
Comment (matipojo) [inline modules/system-info/assets/js/admin/module.js:15]: Add it as an inline scripts.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/usage/usage-reporter.php [hunk 2]
```diff
@@ -91,4 +91,23 @@ public function get_raw_usage() {
 			'value' => $usage,
 		];
 	}
+
+	private function getRemoveRecalcQueryStringScript() {
```
Comment (matipojo) [inline modules/usage/usage-reporter.php:95]: snake_case.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/usage/usage-reporter.php [hunk 3]
```diff
@@ -91,4 +91,23 @@ public function get_raw_usage() {
...
+		?>
+		<script>
+			{
+				const url = new URL( window.location );
+
+				url.hash = '';
+				url.searchParams.delete( 'elementor_usage_recalc' );
+				url.searchParams.delete( '_wpnonce' );
+
+				history.pushState( '', window.title, url.toString() );
```
Comment (matipojo) [inline modules/usage/usage-reporter.php:106]:
```suggestion
				history.replaceState( '', window.title, url.toString() );
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/usage/usage-reporter.php [hunk 4]
```diff
@@ -91,4 +91,23 @@ public function get_raw_usage() {
 			'value' => $usage,
 		];
 	}
+
+	private function get_remove_recalc_query_string_script() {
```
Comment (matipojo) [inline modules/usage/usage-reporter.php:95]: Explain why not use a `redirect` header.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

File: modules/usage/usage-reporter.php [hunk 5]
```diff
@@ -91,4 +91,29 @@ public function get_raw_usage() {
...
 		];
 	}
+
+	/**
+	 * Removes the "elementor_usage_recalc" param from the query string to avoid recalc every refresh.
+	 * When using a redirect header in place of this approach it throws an error because some components have already output some content.
+	 *
+	 * @return string
+	 */
+	private function get_remove_recalc_query_string_script() {
```
Comment (kobizz) [inline modules/usage/usage-reporter.php:101]: Try to find a way to print this script as an external file or inside the general Elementor's 'admin.js'.

**Review by kobizz — CHANGES_REQUESTED** [review]

File: modules/usage/usage-reporter.php (hunk 5 above)
Comment (matipojo) [reply modules/usage/usage-reporter.php:101]: It's not worth the time.

Comment (matipojo) [reply modules/usage/usage-reporter.php:101]: @Nevoss Please add a comment with the file path that creates this script - for easy debugging.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

File: modules/usage/usage-reporter.php [hunk 6]
```diff
@@ -20,14 +20,18 @@ class Usage_Reporter extends Base {
...
-		if ( 'html' === $this->_properties['format'] && empty( $_GET[ self::RECALC_ACTION ] ) ) { // phpcs:ignore -- nonce validation is not require here.
-			$nonce = wp_create_nonce( self::RECALC_ACTION );
-			$url = add_query_arg( [
-				self::RECALC_ACTION => 1,
-				'_wpnonce' => $nonce,
-			] );
-
-			$title .= '<a id="elementor-usage-recalc" href="' . esc_url( $url ) . '#elementor-usage-recalc" class="box-title-tool">Recalculate</a>';
+		if ( 'html' === $this->_properties['format'] ) {
+			if ( empty( $_GET[ self::RECALC_ACTION ] ) ) { // phpcs:ignore -- nonce validation is not require here.
```
Comment (kobizz) [inline modules/usage/usage-reporter.php:24]:
```suggestion
			if ( empty( $_GET[ self::RECALC_ACTION ] ) ) { // phpcs:ignore -- nonce validation is not required here.
```

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**